service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java

```diff
@@ -569,7 +569,19 @@ public TExecuteStatementResp ExecuteStatement(TExecuteStatementReq req) throws T
               queryTimeout) : cliService.executeStatement(sessionHandle, statement, confOverlay,
               queryTimeout);
       resp.setOperationHandle(operationHandle.toTOperationHandle());
-      resp.setStatus(OK_STATUS);
+      Operation operation = cliService.getSessionManager().getOperationManager()
+          .getOperation(operationHandle);
+      if (operation.showOperationDrilldownLink() && cliService.getHiveServer2().getWebServerUrl() != null) {
```

Comment:
I think this should be moved to somewhere around `CliService#executeStatement`

Reply:
CliService defines the executeStatement returning OperationHandle, which has little relationship with thrift interface/definition. It's may not nice to wrap the thrift message here. Thanks for the review, @kgyrtkirk!